sklearn/gaussian_process/kernels.py

```diff
@@ -1141,14 +1153,23 @@ def __call__(self, X, Y=None, eval_gradient=False):
             hyperparameter of the kernel. Only returned when eval_gradient
             is True.
         """
+        length_scale = self.length_scale
+        if np.iterable(length_scale):
+            if len(length_scale) > 1:
+                length_scale = np.asarray(length_scale, dtype=np.float)
+            else:
+                length_scale = float(length_scale[0])
+        else:
+            length_scale = float(length_scale)
+
         X = np.atleast_2d(X)
-        if self.anisotropic and X.shape[1] != self.length_scale.shape[0]:
+        if self.anisotropic and X.shape[1] != length_scale.shape[0]:
             raise Exception("Anisotropic kernel must have the same number of "
```

Comment:
I mean the [else case below](https://github.com/scikit-learn/scikit-learn/pull/6963/files/19d6e2490e25e6670f1b9ab15541a61d0c2052d4#diff-fb7476d7a12141c65a563c589a014cb3R1200) has already been caught out here.